### PR TITLE
Include FastFurnace mod (features).

### PR DESCRIPTION
**Pull request by Oganesson897**
# Summary
Included the [FastFurnace](https://www.curseforge.com/minecraft/mc-mods/fastfurnace) mod.
Not FastWorkbench.

### CODE REVIEW

**Comment by Oregano1963** [comment]
Fast Furnace is superseded by UT. There's no need to bundle it in

**Comment by kappa-maintainer** [comment]
Moving fixes for vanilla directly into Cleanroom is one of our goal
The only problem is, will this PR crash UT's mixin?

**Comment by jchung01** [comment]
> Fast Furnace is superseded by UT. There's no need to bundle it in

I don't think this is true, UT caches crafting recipes only (so that's FastWorkbench functionality?). If I remember correctly, CensoredASM is the one that supersedes FastFurnace, and it mixins into `FurnaceRecipes`, not `TileEntityFurnace`. You might want to check Rong's impl in CASM if it's better than FastFurnace and use that.

**Comment by Oganesson897** [comment]
`FastFurnace` replaced the furnace block, and it will be override the `createNewTileEntity` function to creat the faster furnace tile entity.

so run the game with FF is safety.  
I searched furnace keywords in UT's curse page, but not found any info of FastFurnace Included.

File: patches/minecraft/net/minecraft/tileentity/TileEntityFurnace.java.patch [hunk 1]
```diff
@@ -1,6 +1,40 @@
 --- before/net/minecraft/tileentity/TileEntityFurnace.java
 +++ after/net/minecraft/tileentity/TileEntityFurnace.java
-@@ -37,20 +37,18 @@
+@@ -3,6 +3,7 @@
+ import net.minecraft.block.Block;
```
Comment (Ecdcaeb) [inline patches/minecraft/net/minecraft/tileentity/TileEntityFurnace.java.patch:4]: Use the full class name, don't patch the import
This is a waste of text.

Comment (Oganesson897) [reply patches/minecraft/net/minecraft/tileentity/TileEntityFurnace.java.patch]: Removed

**Comment by Oganesson897** [comment]
I tested the compability this feature with UT or FastFurnace.
Not crash and furnace is workable.

**Comment by Rongmario** [comment]
It's better to use loliasm's implementation than fastfurnace's.

**Comment by Oganesson897** [comment]
> It's better to use loliasm's implementation than fastfurnace's.

Ok

**Pull request closed without merging**